### PR TITLE
oss/util: Add Append function to Nocasemaps

### PR DESCRIPTION
**Pull request by as2643**
Added append functionality to nocasemaps for multiple values at a time, where the key will be lowered and the value will be added to the map.

### CODE REVIEW

**Review by nickoneill — COMMENTED** [review]
Can we back this up with some tests for a couple different types?

File: util/nocasemaps/nocase.go [hunk 1]
```diff
@@ -74,6 +74,23 @@ func Delete[K ~string, V any](m map[K]V, k K) {
...
+	if isLowerASCII(string(k)) {
+		m[k] = append(m[k], vs...)
+		return
+	}
+
+	// if key needs to become lowercase, uses appendToLower
+	var a [stackArraySize]byte
+	s := appendToLower(a[:0], string(k))
+	m[K(s)] = append(m[K(s)], vs...)
+
```
Comment (dsnet) [inline util/nocasemaps/nocase.go:91]: Nit: Remove blank line at the end?

File: util/nocasemaps/nocase.go [hunk 2]
```diff
@@ -74,6 +74,23 @@ func Delete[K ~string, V any](m map[K]V, k K) {
 	delete(m, K(appendToLower(a[:0], string(k))))
 }
 
+// Append is equivalent to:
+//
+//	append(m[strings.ToLower(k)], v)
+func Append[K ~string, S []E, E any](m map[K]S, k K, vs ...E) {
```
Comment (dsnet) [inline util/nocasemaps/nocase.go:80]: Bikeshed: I think `AppendElem` or `AppendSlice` or `AppendSliceElem` is a clearer name as `Append` is not an operation that makes sense for maps.

**Comment by dsnet** [comment]
The commit title should probably be something like:
> util/nocasemaps: add Append method

Of course "Append" may change if we rename it per my comment elsewhere.

**Review by dsnet — APPROVED** [review]
Thanks!

**Review by sailorfrag — APPROVED** [review]

**Pull request merged**